docs/project-brief.md

```diff
@@ -0,0 +1,89 @@
+# Curiostack Project Brief
+
+###### Goals, concepts, challenges, timeline, inspiration
+
+### What is it?
+
+Curiostack is a full-featured framework for creating new web applications and experiences. It is
+designed to not stop at any single need - writing code, setting up monitoring, automatic deployment,
+and any other aspect of product development is covered. Users should be able to use the stack to
+focus on writing business logic for a service, not production / dev-ops, and follow all the best
+practices for free.
+
+### Who is it for?
+
+- Inexperienced developers not yet familiar with development best practices
+- Developers that want to get up-and-running quickly without sacrificing production-readiness
+- Startups with a small development team but large aspirations for scale
+
+### Why build it?
+
+To have a platform for quickly developing Curioswitch services in a production-ready manner.
+
+To allow anyone else to take advantage of the platform to be able to focus on creation rather than
+operations.
+
+### Non-goals
+
+To create a flexible, plugabble stack. Curiostack is heavily opinionated to achieve a simple and
+quick starter experience. Those that already have a good grasp of best practices and have systems
+in place for Curiostack's features don't need it.
+
+To support multiple cloud providers. Currently Curiostack only targets Google Cloud Platform (GCP).
+GCP has many high-quality, scalable features and fits well with the Curiostack goals. AWS support
+would also be interesting but is not currently a priority. If undertaken, it would need to have
+1-1 feature parity with GCP support.
+
+### Team
+
+Choko, and anyone else in the world that would like to contribute.
+
+### Features
+- Simple, generally one-step, commands for setting up a cloud project with a Google Container 
+Engine (GKE) cluster, creating databases, and other common cluster operations.
+- Build system support for setting up client/server API-based services with minimal boilerplate.
+- Automatic test and beta deployment of services.
+- Security first. All endpoints will use HTTPs, and security best practices will be baked in without
+the need for user configuration.
+
+### Challenges and open questions
+
+- A more thorough design document is needed for probing the technical aspects of Curiostack.
+- Is there a finer balance needed between easy-to-use pre-baked opinions and user configurability?
+- Is the minimum cost of a Google Container Engine cluster too high for experimentation?
```

Comment:
IMO, yes. Especially, the new users will have a concern whether this edge tech really could help them to build a stable service for their startup career. While maybe if we could using this stack build some awesome on-line applications, then this problem will be resolved.